opennms-correlation/drools-correlation-engine/src/test/java/org/opennms/netmgt/correlation/drools/FusionRulesIT.java

```diff
@@ -0,0 +1,175 @@
+
+
+@RunWith(OpenNMSJUnit4ClassRunner.class)
+@ContextConfiguration(locations={
+        "classpath:/META-INF/opennms/applicationContext-daemon.xml",
+        "classpath:/META-INF/opennms/applicationContext-commonConfigs.xml",
+        "classpath:/META-INF/opennms/applicationContext-minimal-conf.xml",
+        "classpath:/META-INF/opennms/applicationContext-dao.xml",
+        "classpath*:/META-INF/opennms/component-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-soa.xml",
+        "classpath:/META-INF/opennms/mockEventIpcManager.xml",
+        "classpath:/META-INF/opennms/correlation-engine.xml",
+        "classpath:/META-INF/opennms/applicationContext-alarmd.xml",
+        "classpath:/test-context.xml"
+})
+@JUnitConfigurationEnvironment(systemProperties={"org.opennms.activemq.broker.disable=true"})
+@JUnitTemporaryDatabase(dirtiesContext=false,tempDbClass= MockDatabase.class,reuseDatabase=false)
+public class FusionRulesIT implements TemporaryDatabaseAware<MockDatabase>, InitializingBean {
+
+
+    @Autowired
+    private MockEventIpcManager m_eventIpcMgr;
+
+    @Autowired
+    private CorrelationEngineRegistrar m_correlator;
+
+    @Autowired
+    private AlarmDao m_alarmDao;
+
+    @Autowired
+    private Alarmd m_alarmd;
+
+    private MockDatabase m_database;
+
+    protected Integer m_anticipatedMemorySize = 0;
+
+
+    public FusionRulesIT() {
+        ConfigurationTestUtils.setRelativeHomeDirectory("src/test/opennms-home");
+    }
+
+    @Before
+    public void setup() {
+        m_eventIpcMgr.setEventWriter(m_database);
+    }
+
+    @Override
+    public void afterPropertiesSet() {
+        BeanUtils.assertAutowiring(this);
+    }
+
+    public void setCorrelationEngineRegistrar(CorrelationEngineRegistrar correlator) {
+        m_correlator = correlator;
+    }
+
+    @Test
+    public void testDroolsFusion() throws Exception {
+
+        anticipate(createNodeUpEvent(1));
+        anticipate(new EventBuilder(EventConstants.RELOAD_DAEMON_CONFIG_SUCCESSFUL_UEI, "droolsFusion").getEvent());
+        anticipate(new EventBuilder(EventConstants.DROOLS_ENGINE_ENCOUNTERED_EXCEPTION, "droolsFusion").getEvent());
+        EventBuilder eb = new EventBuilder("uei.opennms.org/triggerTestForFusion", "droolsTest");
+        DroolsCorrelationEngine engine = findEngineByName("droolsFusion");
+        // This will create an exception and the engine should reload.
+        engine.correlate(eb.getEvent());
+        Thread.sleep(10000); // Wait time for the engine to reload.
+        // The engine should run other rules after reload.
+        engine.correlate(createNodeLostServiceEvent(1, "SSH"));
+        Thread.sleep(4000); // Give time to execute the time-based rules.
+        m_anticipatedMemorySize = 5;
+        engine.getKieSessionObjects().forEach(System.err::println);
+        verify(engine);
+        OnmsAlarm alarm = m_alarmDao.findByReductionKey(EventConstants.DROOLS_ENGINE_ENCOUNTERED_EXCEPTION + ":" + "droolsFusion");
```

Comment:
I think it would be sufficient to verify that the event is generated when an exception occurs - we know alarm handling works, so there's no need to verify that explicitly here.